dynamiqs/integrators/apis/floquet.py

```diff
@@ -174,10 +174,13 @@ def _check_floquet_args(
     )
 
     # === check that the Hamiltonian is periodic with the supplied period
-    H = eqx.error_if(
-        H,
-        jnp.logical_not(jnp.isclose(H(0), H(T))),
-        'The Hamiltonian H is not periodic with the supplied period T.',
-    )
+    if not isinstance(H, CallableTimeArray):
+        # disable the check for CallableTimeArray as equinox cannot find an
+        # underlying array to attach the error message to.
+        H = eqx.error_if(
+            H,
+            jnp.logical_not(jnp.isclose(H(0.0)._data, H(T)._data)),  # noqa: SLF001
+            'The Hamiltonian H is not periodic with the supplied period T.',
+        )
```

Comment:
This is a little strange? Even if `H` has type `CallableTimeArray` shouldn't e.g. `H(0.0)` return a `QArray`? What am I missing here?

Reply:
It's because of the way the `eqx.error_if` runtime error works. The second argument `jnp.logical_not(jnp.isclose(H(0.0)._data, H(T)._data))` works as expected, but the first argument `H` is the faulty one here. Indeed, I think `eqx.error_if` requires `H` to be a PyTree with at least one underlying array so it can "attach" the error message to this array. But a `TimeCallable` only has a callable as a leaf, and it doesn't find the a corresponding array, leading to an error.

There might be workarounds, but I couldn't find one.

Reply:
I see, understood. Indeed I see in the `error_if` docs that the PyTree passed to this must have at least one array.

Reply:
Otherwise everything else LGTM :)

Reply:
Actually maybe I can replace `H` with `H(0.0)` here, not sure, will try...

Reply:
With the code as is I don't think that will work, since as I understand it the PyTree on the left gets replaced by the PyTree in the first argument to `error_if`. The following also would not work
```suggestion
    if not isinstance(H, CallableTimeArray):
        # disable the check for CallableTimeArray as equinox cannot find an
        # underlying array to attach the error message to.
        H0 = eqx.error_if(
            H(0.0),
            jnp.logical_not(jnp.isclose(H(0.0)._data, H(T)._data)),  # noqa: SLF001
            'The Hamiltonian H is not periodic with the supplied period T.',
        )
```
since as I understand it, as `H0` is not used later, this code is eliminated under dead-code elimination https://docs.kidger.site/equinox/api/errors/. Seems a little tricky and annoying and any workarounds feel hackish...

Reply:
Fixed with 8513423786eac60c1157fac97acd7d84d45ca599.